sycl/source/detail/scheduler/commands.cpp

```diff
@@ -230,6 +230,47 @@ Command::getPiEvents(const std::vector<EventImplPtr> &EventImpls) const {
   return RetPiEvents;
 }
 
+// This function is implemented (duplicating getPiEvents a lot) as short term
+// solution for the issue that shortcut functions bypass graph and could not
+// handle empty pi event handles when kernel is enqueued on host task
+// completion. Used for barrier with wait list.
```

Comment:
If I understand it correctly, this is a barrier specific issue, not a shortcut one. The barrier shortcut is mapped to the full handler version, so I would expect both to run into the same issue. And they don't really bypass the graph in the same manner as the USM shortcuts (the node is still created, it's just isolated), so it probably makes sense to make that difference clearer.

Reply:
done